interceptor/copy_read_closer.go

```diff
@@ -42,7 +43,12 @@ type copyReadCloser struct {
 // Second read after EOF
 // copyBuffered --> copy BufReader simple buffer with fix size
 // when BufReader is EOF offset is reset to read again
-func NewCopyReadCloser(src io.ReadCloser) *copyReadCloser {
+func NewCopyReadCloser(src io.ReadCloser) io.ReadCloser {
```

Comment:
For the reviewers :

- We now return an `io.ReadCloser` interface instead of a concrete `copyReadCloser`.
- When the request body is nil, the Interceptor middleware/tripperware segfaults as it tries to read a nil body without checking it.